### ISSUE TITLE
`import_playbook` does no more ignore `git-crypt` locked files defined in `.config/ansible-lint.yml`

### ISSUE THREAD

**Issue by macwinnie** [ansible/ansible-lint#2987]
##### Summary

Since version v6.12.1 seams to follow the imports, `git-crypt` locked files defined in `.config/ansible-lint.yml` are no more ignored properly. But only if `import_playbook` is mixed with direct playbook definitions.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

GitHub Actions and `pre-commit` repo usage with latest version

```
v6.12.1
```

- ansible installation method: pip
- ansible-lint installation method: pip and GitHub Action

##### STEPS TO REPRODUCE

We've got a playbook `/run.yml` that is built up like that:

```yaml
- hosts: all
  become: true
  tasks:
    - name: install vim
      ansible.builtin.apt:
        name: vim

- ansible.builtin.import_playbook: playbooks/users.yml
```

The Playbook `/playbooks/users.yml` calls some roles – especially the `users` one with a `secrets.yml` located in the `roles/users/defaults/main/secrets.yml` path. That file is locked by `git-crypt`.

The `/.config/ansible-lint.yml` file has this exception:

```yaml
exclude_paths:
  - ../.*
  - ../**/secret*.y*ml
```

##### Desired Behavior

The `secrets.yml` should be ignored due to the `/.config/ansible-lint.yml` definition.

##### Actual Behavior

Throwing an error:

```paste below
WARNING  Listing 1 violation(s) that are fatal
internal-error: Unexpected error code 250 from execution of: ansible-playbook --syntax-check -i localhost, run.yml
run.yml:1 ERROR! Unexpected Exception, this is probably a bug: 'utf-8' codec can't encode character '\udcc9' in position 10: surrogates not allowed
```

##### Hot Fix on our side

When the `task` part also is moved to a separate playbook, everything is well. So an additional playbook `/playbooks/basic.yml` and the changed file below are working fine.

`playbooks/basic.yml`:

```yaml
- hosts: all
  become: true
  tasks:
    - name: install vim
      ansible.builtin.apt:
        name: vim
```

`run.yml`:

```yaml
- ansible.builtin.import_playbook: playbooks/basic.yml
- ansible.builtin.import_playbook: playbooks/users.yml
```

**Comment by ssbarnea** [comment]
If ansible-playbook --syntax-check fails, ansible-lint will fail too and adding excludes will not help, especially as the file causing the issue is not the "crypted" file, is the one doing the inclusion.

If you add the `run.yml` to exclude, probably you will not see that error, but you will also not lint it.
